packages/api/src/share_link/service.ts

```diff
@@ -69,4 +74,141 @@ const createShareLink = async (
+  }
+  if (requestData.accessRestrictions !== undefined) {
+    columnsToUpdate.unrestricted = requestData.accessRestrictions === "none";
+    columnsToUpdate.autoapprovetoggle =
+      requestData.accessRestrictions === "approval" ? 0 : 1;
+  }
+  if (requestData.maxUses !== undefined) {
+    columnsToUpdate.maxuses = requestData.maxUses;
+  }
+  if (requestData.expirationTimestamp !== undefined) {
+    columnsToUpdate.expiresdt = requestData.expirationTimestamp;
+  }
+  return columnsToUpdate;
+};
+
+const buildPatchQuery = (
+  requestData: UpdateShareLinkRequest,
+  shareLinkId: string
+): {
+  query: string;
+  params: UpdateableShareLinkColumns | { shareLinkId: string; email: string };
+} => {
+  const columnsForUpdate = getShareLinkColumnValuesForUpdate(requestData);
+  const updates = Object.entries(columnsForUpdate)
+    .filter(([_, value]) => value !== undefined)
+    .map(([key, _]) => `${key} = :${key}`);
+
+  if (updates.length === 0) {
+    throw new createError.BadRequest("Request cannot be empty");
+  }
+
+  const query = `
+    UPDATE
+      shareby_url
+    SET ${updates.join(", ")}
+    FROM
+      account, folder_link
+    WHERE
+      shareby_urlid = :shareLinkId
+      AND account.accountid = shareby_url.byaccountid
+      AND account.primaryEmail = :email
+      AND folder_link.folder_linkid = shareby_url.folder_linkid
+    RETURNING
+    shareby_url.shareby_urlid AS "id",
+    COALESCE(folder_link.recordid, folder_link.folderid) AS "itemId",
+    CASE
+      WHEN folder_link.recordid IS NOT NULL THEN 'record'
+      ELSE 'folder'
+    END AS "itemType",
+    shareby_url.urltoken AS "token",
+    substring(shareby_url.defaultaccessrole FROM (length('access.role.') + 1)) AS "permissionsLevel",
+    CASE
+      WHEN shareby_url.unrestricted THEN 'none'
+      WHEN shareby_url.autoapprovetoggle = 0 THEN 'approval'
+      ELSE 'account'
+    END AS "accessRestrictions",
+    CASE
+      WHEN shareby_url.maxuses = 0 THEN NULL
+      ELSE shareby_url.maxuses
+    END AS "maxUses",
+    shareby_url.uses AS "usesExpended",
+    shareby_url.expiresdt AS "expirationTimestamp",
+    shareby_url.createddt AS "createdAt",
+    shareby_url.updateddt AS "updatedAt";
+  `;
+
+  return {
+    query: query.trim(),
+    params: {
+      ...columnsForUpdate,
+      shareLinkId,
+      email: requestData.emailFromAuthToken,
+    },
+  };
+};
+
+const updateShareLink = async (
+  shareLinkId: string,
+  data: UpdateShareLinkRequest
+): Promise<ShareLink> => {
+  const shareLinkResult = await db
+    .sql<ShareLink>("share_link.queries.get_share_link", {
+      shareLinkId,
+      email: data.emailFromAuthToken,
+    })
+    .catch((err) => {
+      logger.error(err);
+      throw new Error("Failed to get share link");
+    });
+
+  if (shareLinkResult.rows[0] === undefined) {
+    throw new createError.NotFound("Share link not found");
+  }
+
+  if (
+    (shareLinkResult.rows[0].accessRestrictions === "none" &&
```

Comment:
This if block is pretty hard to parse through; would it be possible to break this into a few named helper functions (can be define in this file / right above).  The purpose of the function wouldn't be re-usability necessarily, but rather legibility.